lib/mongo/auth/scram/conversation.rb

```diff
@@ -247,6 +255,8 @@ def client_final
         # @since 2.0.0
         def client_key
           @client_key ||= hmac(salted_password, CLIENT_KEY)
+          user.client_key ||= @client_key
+          @client_key
```

Comment:
After changing User#client_key to be private, I can't use `||=` anymore and instead need to do `user.send(:client_key=, @client_key) unless user.send(:client_key)`, so I've left it in.